modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/api/ItSqlApiBaseTest.java

```diff
@@ -0,0 +1,860 @@
+        checkError(
+                ColumnNotFoundException.class,
+                ErrorGroups.Table.COLUMN_NOT_FOUND_ERR,
+                "Column does not exist [tableName=\"PUBLIC\".\"TEST\", columnName=\"VAL1\"]",
+                ses,
+                "ALTER TABLE TEST DROP COLUMN VAL1"
+        );
+
+        // DROP TABLE
+        checkDdl(false, ses, "DROP TABLE IF EXISTS NOT_EXISTS_TABLE");
+
+        checkDdl(true, ses, "DROP TABLE TEST");
+        checkError(
+                TableNotFoundException.class,
+                ErrorGroups.Table.TABLE_NOT_FOUND_ERR,
+                "The table does not exist [name=\"PUBLIC\".\"TEST\"]",
+                ses,
+                "DROP TABLE TEST"
+        );
+
+        checkDdl(false, ses, "DROP INDEX IF EXISTS TEST_IDX");
+
+        checkError(
+                IndexNotFoundException.class,
+                Index.INDEX_NOT_FOUND_ERR,
+                "Index does not exist [name=\"PUBLIC\".\"TEST_IDX\"]", ses,
+                "DROP INDEX TEST_IDX"
+        );
+    }
+
+    /** Check all transactions are processed correctly even with case of sql Exception raised. */
+    @Test
+    public void implicitTransactionsStates() {
+        IgniteSql sql = igniteSql();
+
+        if (sql instanceof ClientSql) {
+            return;
+        }
+
+        sql("CREATE TABLE TEST(ID INT PRIMARY KEY, VAL0 INT)");
+
+        Session ses = sql.createSession();
+
+        TxManager txManager = txManager();
+
+        for (int i = 0; i < ROW_COUNT; ++i) {
+            assertThrowsSqlException(
+                    Catalog.VALIDATION_ERR,
+                    "Table with name 'PUBLIC.TEST' already exists",
+                    () -> execute(ses, "CREATE TABLE TEST(ID INT PRIMARY KEY, VAL0 INT)")
+            );
+        }
+
+        // No new transactions through ddl.
+        assertEquals(0, txManager.pending());
+    }
+
+    /** Check correctness of implicit and explicit transactions. */
+    @Test
+    public void checkTransactionsWithDml() {
+        IgniteSql sql = igniteSql();
+
+        if (sql instanceof ClientSql) {
+            return;
+        }
+
+        sql("CREATE TABLE TEST(ID INT PRIMARY KEY, VAL0 INT)");
+
+        Session ses = sql.createSession();
+
+        TxManager txManagerInternal = txManager();
+
+        int txPrevCnt = txManagerInternal.finished();
+
+        for (int i = 0; i < ROW_COUNT; ++i) {
+            checkDml(1, ses, "INSERT INTO TEST VALUES (?, ?)", i, i);
+        }
+
+        // Outer tx with further commit.
+        Transaction outerTx = igniteTx().begin();
+
+        for (int i = ROW_COUNT; i < 2 * ROW_COUNT; ++i) {
+            checkDml(1, outerTx, ses, "INSERT INTO TEST VALUES (?, ?)", i, i);
+        }
+
+        commit(outerTx);
+
+        // Outdated tx.
+        Transaction outerTx0 = outerTx;
+        //ToDo: IGNITE-20387 , here should be used assertThrowsSqlException method with code and message `"Transaction is already finished"
```

Comment:
Apparently there should be a link to another ticket? :thinking:

Reply:
no, it has been removed )